src/main/resources/application.properties

```diff
@@ -5,6 +5,8 @@ spring.datasource.url=jdbc:postgresql://${DATABASE_ENDPOINT}:${DATABASE_PORT}/${
 spring.datasource.username=${DATABASE_USERNAME}
 spring.datasource.password=${DATABASE_PASSWORD}
 spring.datasource.driverClassName=org.postgresql.Driver
+spring.datasource.tomcat.test-while-idle=true
+spring.datasource.tomcat.validation-query=SELECT 1
```

Comment:
Would it be helpful to turn on the sql logging to verify that the above validation query runs prior to the timeout? Unless there's another way that indicates the connection is being kept alive...